react_ujs/index.js

```diff
@@ -108,7 +118,8 @@ var ReactRailsUJS = {
       var className = node.getAttribute(ujs.CLASS_NAME_ATTR);
       var constructor = ujs.getConstructor(className);
       var propsJson = node.getAttribute(ujs.PROPS_ATTR);
-      var props = propsJson && JSON.parse(propsJson);
+      var props = propsJson && (ujs.options.replaceNull ? replaceNullWithUndefined(JSON.parse(propsJson))
```

Comment:
why not a separate var for the parse result to avoid duplicating that bit?

Reply:
If the ruby solution is good, all js implementations will be removed.